### PR TITLE
ronin: init at 2.0.1

### PR DESCRIPTION
**Pull request by Ch1keen**
###### Description of changes

Add [ronin](https://ronin-rb.dev) to nixpkgs.

[Ronin](https://ronin-rb.dev) is a free and Open Source Ruby toolkit for security research and development. Ronin also allows for the rapid development and distribution of code, exploits, payloads, etc, via 3rd party git repositories. 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -2590,6 +2590,12 @@
       }
     ];
   };
+  Ch1keen = {
+    email = "gihoong7@gmail.com";
+    github = "Ch1keen";
+    githubId = 40013212;
+    name = "Han Jeongjun";
+  };
```
Comment (ocfox) [inline maintainers/maintainer-list.nix:2598]: Please add maintainer in a separate commit.

Comment (Ch1keen) [reply maintainers/maintainer-list.nix:2598]: Should I rebase the commit?

Comment (ocfox) [reply maintainers/maintainer-list.nix:2598]: No, split into two commits, see https://github.com/NixOS/nixpkgs/pull/222982/commits

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -39810,4 +39810,6 @@ with pkgs;
   zf = callPackage ../tools/misc/zf { };
 
   isolate = callPackage ../tools/security/isolate { };
+
+  ronin = callPackage ../tools/security/ronin { };
```
Comment (ocfox) [inline pkgs/top-level/all-packages.nix:39814]: Place in alphabetical order in the approximate position.

**Comment by ocfox** [comment]
Please squash into two commits.

File: pkgs/tools/security/ronin/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    "ronin-exploits"
+    "ronin-fuzzer"
+    "ronin-payloads"
+    "ronin-repos"
+    "ronin-vulns"
+    "ronin-web"
+  ];
+
+  meta = with lib; {
+    description = "Ronin is a free and Open Source Ruby toolkit for security research and development";
```
Comment (Ch1keen) [inline pkgs/tools/security/ronin/default.nix:20]: Should I replace this line with "Free and Open Source Ruby toolkit for security research and development"?

File: pkgs/tools/security/ronin/default.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
+{ pkgs, lib, bundlerApp }:
+
+bundlerApp {
+  pname = "ronin";
+  gemdir = ./.;
+  # ronin-rb requires ruby >= 3.0
+  ruby = pkgs.ruby_3_1;
```
Comment (marsam) [inline pkgs/tools/security/ronin/default.nix:7]: Ruby 3.1 is the default
```suggestion
{ lib, bundlerApp }:

bundlerApp {
  pname = "ronin";
  gemdir = ./.;
```

File: pkgs/tools/security/ronin/default.nix [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    "ronin"
+    "ronin-db"
+    "ronin-exploits"
+    "ronin-fuzzer"
+    "ronin-payloads"
+    "ronin-repos"
+    "ronin-vulns"
+    "ronin-web"
+  ];
+
```
Comment (marsam) [inline pkgs/tools/security/ronin/default.nix:18]:
```suggestion

  passthru.updateScript = bundlerUpdateScript "ronin";

```

File: pkgs/tools/security/ronin/default.nix (hunk 3 above)
Comment (marsam) [reply pkgs/tools/security/ronin/default.nix:20]: Yes please. In general the `description` field should not include the package name.

File: pkgs/tools/security/ronin/.bundle/config [hunk 6]
```diff
@@ -0,0 +1,2 @@
+---
+BUNDLE_FORCE_RUBY_PLATFORM: "true"
```
Comment (marsam) [inline pkgs/tools/security/ronin/.bundle/config:2]: Please remove this file since it doesn't have any effect in the package.

**Review by marsam — APPROVED** [review]

**Pull request merged**